rule/redundant_test_main_exit.go

```diff
@@ -0,0 +1,75 @@
+package rule
+
+import (
+	"fmt"
+	"go/ast"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// RedundantTestMainExitRule suggests removing Exit call in TestMain function for test files.
+type RedundantTestMainExitRule struct{}
+
+// Apply applies the rule to given file.
+func (*RedundantTestMainExitRule) Apply(file *lint.File, _ lint.Arguments) []lint.Failure {
+	var failures []lint.Failure
+
+	onFailure := func(failure lint.Failure) {
+		failures = append(failures, failure)
+	}
+
+	w := &lintRedundantTestMainExit{onFailure: onFailure, isTestFile: file.IsTest()}
+	ast.Walk(w, file.AST)
+	return failures
+}
+
+// Name returns the rule name.
+func (*RedundantTestMainExitRule) Name() string {
+	return "redundant-test-main-exit"
+}
+
+type lintRedundantTestMainExit struct {
+	onFailure  func(lint.Failure)
+	isTestFile bool
+}
+
+func (w *lintRedundantTestMainExit) Visit(node ast.Node) ast.Visitor {
+	if !w.isTestFile {
+		return nil // skip analysis of this file if it is not a test file
+	}
+
+	se, ok := node.(*ast.ExprStmt)
+	if !ok {
+		return w
+	}
+	ce, ok := se.X.(*ast.CallExpr)
+	if !ok {
+		return w
+	}
+
+	fc, ok := ce.Fun.(*ast.SelectorExpr)
+	if !ok {
+		return w
+	}
+	id, ok := fc.X.(*ast.Ident)
+	if !ok {
+		return w
+	}
+
+	pkg := id.Name
+	fn := fc.Sel.Name
+	if isCallToExitFunction(pkg, fn) {
```

Comment:
`isCallToExitFunction` will return `true` even for function calls other than `os.Exit` (for example `log.Fatal`, `log.Panic` ...) That behavior is okay for what we check here?

Reply:
Hi @chavacava, thank you for your detailed review and suggestions!

I’m not entirely sure about including logging functions other than `os.Exit` or `syscall.Exit`. It’s uncommon (in my experience) to call logging functions that internally invoke Exit, but I don't see any downside to including them in this check.

What’s your opinion on this?

Reply:
The issue with including logging functions is that the common failure message may not be appropriate for them, as these functions are not necessarily redundant: they first log a message and then exit.